src/base/lineparser.cpp

```diff
@@ -585,8 +584,9 @@ LineParser::ParseReturnValue LineParser::readNextLine(int optionMask)
     // Broadcast result of file check
     if (processPool_)
     {
-        if (!processPool_->broadcast(EnumCast<LineParser::ParseReturnValue>(result)))
-            return LineParser::Fail;
+        int enumValue = getIntFromParseReturnValue(result);
+        processPool_->broadcast(enumValue);
```

Comment:
Should check the broadcast result as well:
```suggestion
        if (!processPool_->broadcast(enumValue))
            return LineParser::Fail;
```